torch/_refs/__init__.py

```diff
@@ -2224,7 +2224,6 @@ def _fn(*args, out=None, **kwargs):
 py_all = all
 
 
-@register_decomposition(aten.all)
 @out_wrapper()
 def all(
```

Comment:
Since this isn't calling a `prims.all`, we should update the reference instead of duplicating it as a decomposition. Decomposing into `any` makes a lot more sense than `sum` anyway, so this seems good.

Reply:
I tried updating the refs to use the new decomposition, but it seems there is an issue with using `aten` ops in `_refs` implementations:

```
  File "/home/ssjia/Github/pytorch/torch/_refs/__init__.py", line 2235, in all
    result = aten.logical_not(aten.any(aten.logical_not(a)))
                                       ^^^^^^^^^^^^^^^^^^^
  File "/home/ssjia/Github/pytorch/torch/_ops.py", line 742, in __call__
    return self._op(*args, **kwargs or {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ssjia/Github/pytorch/torch/_prims/context.py", line 141, in __torch_function__
    raise RuntimeError(
RuntimeError: no _refs support for aten.logical_not
```

For decompositions included in the core ATen decomp table, we want to exclusively decompose to other ATen ops, i.e. avoid decomposing to prim ops, hence why ATen ops are referenced explicitly in the decomposition.

In this case, do you think it would be preferable to keep the ref implementation and the core decomposition separate? Or do you think it might be better to remove the ref implementation in favor of the core implementation?

Reply:
You should be able to call `torch.logical_not` and `torch.any`, or even the methods.

Reply:
thanks for the tip! let me try that...

Reply:
Updated to change the ref implementation instead of introducing a new decomp (y)